cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/types/HasType.kt

```diff
@@ -23,91 +23,116 @@
  *                    \______/ \__|       \______/
  *
  */
-package de.fraunhofer.aisec.cpg.graph
+package de.fraunhofer.aisec.cpg.graph.types
 
 import de.fraunhofer.aisec.cpg.TypeManager
 import de.fraunhofer.aisec.cpg.frontends.TranslationException
-import de.fraunhofer.aisec.cpg.graph.types.Type
-import java.util.Optional
+import de.fraunhofer.aisec.cpg.graph.ContextProvider
+import de.fraunhofer.aisec.cpg.graph.Node
+import de.fraunhofer.aisec.cpg.graph.declarations.ValueDeclaration
+import java.util.*
 
 interface HasType : ContextProvider {
+
```

Comment:
I understand the difference between `declaredType` and `assignedType` but I keep wondering what `type` should be used for

Reply:
I will probably merge declaredType with type or type is an alias for declared type? not sure yet...